src/test/ensemble_flow_custom_node_tests.cpp

```diff
@@ -0,0 +1,974 @@
+        this->prepareRequest(inputValues);
+        auto input_node = std::make_unique<EntryNode>(&request);
+        auto output_node = std::make_unique<ExitNode>(&response);
+        auto custom_node = std::make_unique<CustomNode>(
+            customNodeName,
+            createLibraryMock<T>(),
+            parameters_t{});
+
+        auto pipeline = std::make_unique<Pipeline>(*input_node, *output_node);
+        pipeline->connect(*input_node, *custom_node, {{pipelineInputName, customNodeInputName}});
+        pipeline->connect(*custom_node, *output_node, {{customNodeOutputName, pipelineOutputName}});
+
+        pipeline->push(std::move(input_node));
+        pipeline->push(std::move(custom_node));
+        pipeline->push(std::move(output_node));
+        return pipeline;
+    }
+
+    template <typename T>
+    void checkResponse(std::vector<T> data, std::function<T(T)> op) {
+        this->checkResponse(this->pipelineOutputName, data, op);
+    }
+
+    template <typename T>
+    void checkResponse(const std::string& outputName, std::vector<T> data, std::function<T(T)> op) {
+        this->checkResponse(outputName, this->response, data, op);
+    }
+
+    template <typename T>
+    void checkResponse(const std::string& outputName, const PredictResponse& response, std::vector<T> data, std::function<T(T)> op) {
+        std::transform(data.begin(), data.end(), data.begin(), op);
+        ASSERT_TRUE(response.outputs().contains(outputName));
+        const auto& proto = response.outputs().at(outputName);
+
+        ASSERT_EQ(proto.tensor_content().size(), data.size() * sizeof(T));
+        ASSERT_EQ(proto.tensor_shape().dim_size(), 2);
+        ASSERT_EQ(proto.tensor_shape().dim(0).size(), 1);
+        ASSERT_EQ(proto.tensor_shape().dim(1).size(), data.size());
+
+        auto* ptr = reinterpret_cast<const T*>(proto.tensor_content().c_str());
+
+        const std::vector<T> actual(ptr, ptr + data.size());
+
+        for (size_t i = 0; i < actual.size(); i++) {
+            EXPECT_NEAR(actual[i], data[i], 0.001);
+        }
+    }
+
+    template <typename T>
+    static NodeLibrary createLibraryMock() {
+        return NodeLibrary{
+            T::execute,
+            T::releaseBuffer,
+            T::releaseTensors};
+    }
+
+    PredictRequest request;
+    PredictResponse response;
+
+    NodeLibrary library;
+
+    const std::string customNodeName = "add_sub_node";
+    const std::string libraryName = "add_sub_lib";
+    const std::string libraryPath = "/ovms/bazel-bin/src/lib_node_add_sub.so";
+    const std::string customNodeInputName = "input_numbers";
+    const std::string customNodeOutputName = "output_numbers";
+    const std::string pipelineInputName = "pipeline_input";
+    const std::string pipelineOutputName = "pipeline_output";
+};
+
+TEST_F(EnsembleFlowCustomNodePipelineExecutionTest, AddSubCustomNode) {
+    // Most basic configuration, just process single add-sub custom node pipeline request
```

Comment:
Why couldn't we use `prepareSingleNodePipelineWithLibraryMock` here?

Reply:
We already have add-sub custom node compiled for unit tests. Why not to use it? If we wanted to use `prepareSingleNodePipelineWithLibraryMock`, we would need to rewrite that library to unit test code, ending up with duplicated code.